csharp/Platform.Collections/Lists/IListExtensions.cs

```diff
@@ -343,6 +380,17 @@ public static T[] SkipFirst<T>(this IList<T> list, int skip)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static IList<T> ShiftRight<T>(this IList<T> list) => list.ShiftRight(1);
 
+        /// <summary>
+        /// <para>Shifts all elements of the list to the right by the specified number of elements and returns an array.</para>
+        /// <para>Сдвигает вправо все элементы списка на указанное количество элементов и возвращает массив.</para>
+        /// </summary>
+        /// <param name="list"><para>Copy List.</para><para>Список для копирования.</para></param>
+        /// <param name="skip"><para>Number of items to shift.</para><para>Количество сдвигаемых элементов.</para></param>
+        /// <typeparam name="T"><para>The list's item type.</para><para>Тип элементов списка.</para></typeparam>
+        /// <returns>
+        /// <para>If the variable shift is less than zero - an error is returned, but if the variable shift is 0 an empty array is returned. Otherwise, an array is returned with the shift of the elements.</para>
```

Comment:
`an error is returned` should be replaced with `an <see cref="NotImplementedException"/> exception is thrown`. `an empty array is returned` must be replaced with `an exact copy of the array is returned`.